### PR TITLE
Add KeyguardClient

### PR DESCRIPTION
**Pull request by sisou**
Adds the KeyguardClient code into the `/client` directory.

Open `/client/index.html` in your browser (likely a webserver in the nimiq.github.io root is required).

### CODE REVIEW

File: client/RpcClient.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+     * @param {string} targetOrigin
+     * @returns {Newable}
+     * @private
+     */
+    static _generateClientClass(targetWindow, targetOrigin) {
+        /** @type {Newable} */
+        const Client = class {
+            constructor() {
+                // Svub: Code smell that _targetWindow and _waiting are visible outside. TODO later!
+                this._targetWindow = targetWindow;
```
Comment (danimoh) [inline client/RpcClient.js]: there is not really a need for _targetWindow and _targetOrigin and the fix is easy? so why todo later?

File: client/KeyguardClient.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            type: type || EncryptionType.HIGH,
+            label,
+        }]);
+    }
+
+    /**
+     * @returns {Promise<void>}
+     */
+    async importWords() {
+        return this._popup('import-words');
```
[the remark this answers is not in the document]
Comment (sisou) [reply client/KeyguardClient.js]: I'm only using the names once. As soon as we will use them more often, I'll change it ;)

File: client/KeyguardClient.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
+class KeyguardClient {
+    constructor() {
+        this._keyguardSrc = '../src/';
+
+        this._connected = new Promise(res => {
+            this._connectedResolver = res;
+        });
+
+        this._startIFrame();
```
[the remark this answers is not in the document]
Comment (sisou) [reply client/KeyguardClient.js]: By using a promise as `this._connected` it is possible to await it e.g. in the `list()` method. Thus the client does not have to know about the internal setup and can directly call the methods.

File: client/KeyguardClient.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        // FIXME Remove after debugging
+        this.popup = rpcClient;
+
+        try {
+            const result = await rpcClient.call('request', args);
+            rpcClient.close();
+            $popup.close();
+            return result;
+        } catch (e) {
+            rpcClient.close();
```
[the remark this answers is not in the document]
Comment (sisou) [reply client/KeyguardClient.js]: added

**Pull request merged**